### PR TITLE
Update What's new in .NET 8 for Preview 4

### LINKED ISSUE #34555: Update What's new in .NET 8 for Preview 4
See https://github.com/dotnet/core/issues/8234.

### PR DESCRIPTION
**Pull request by gewarren**
Fixes #34555.

---

#### Internal previews

| 📄 File | 🔗 Preview link |
|:--|:--|
| [docs/core/tools/dotnet-build.md](https://github.com/dotnet/docs/blob/c2ea2364768e6b79b905c4879c0e623fb5219e50/docs/core/tools/dotnet-build.md) | [dotnet build](https://review.learn.microsoft.com/en-us/dotnet/core/tools/dotnet-build?branch=pr-en-us-35340) |
| [docs/core/tools/dotnet-new.md](https://github.com/dotnet/docs/blob/c2ea2364768e6b79b905c4879c0e623fb5219e50/docs/core/tools/dotnet-new.md) | [dotnet new &lt;TEMPLATE&gt;](https://review.learn.microsoft.com/en-us/dotnet/core/tools/dotnet-new?branch=pr-en-us-35340) |
| [docs/core/tools/dotnet-restore.md](https://github.com/dotnet/docs/blob/c2ea2364768e6b79b905c4879c0e623fb5219e50/docs/core/tools/dotnet-restore.md) | [docs/core/tools/dotnet-restore](https://review.learn.microsoft.com/en-us/dotnet/core/tools/dotnet-restore?branch=pr-en-us-35340) |
| [docs/core/whats-new/dotnet-8.md](https://github.com/dotnet/docs/blob/c2ea2364768e6b79b905c4879c0e623fb5219e50/docs/core/whats-new/dotnet-8.md) | [What's new in .NET 8](https://review.learn.microsoft.com/en-us/dotnet/core/whats-new/dotnet-8?branch=pr-en-us-35340) |
| [docs/standard/serialization/system-text-json/how-to.md](https://github.com/dotnet/docs/blob/c2ea2364768e6b79b905c4879c0e623fb5219e50/docs/standard/serialization/system-text-json/how-to.md) | [How to serialize and deserialize (marshal and unmarshal) JSON in .NET](https://review.learn.microsoft.com/en-us/dotnet/standard/serialization/system-text-json/how-to?branch=pr-en-us-35340) |

### CODE REVIEW

File: docs/core/tools/dotnet-build.md [hunk 1]
```diff
@@ -1,7 +1,7 @@
 ---
 title: dotnet build command
 description: The dotnet build command builds a project and all of its dependencies.
-ms.date: 01/23/2023
+ms.date: 05/12/2023
```
Comment (BillWagner) [inline docs/core/tools/dotnet-build.md:4]: Do you want to update the date to the preview 4 release date?

File: docs/core/tools/dotnet-restore.md [hunk 2]
```diff
@@ -1,7 +1,7 @@
 ---
 title: dotnet restore command
 description: Learn how to restore dependencies and project-specific tools with the dotnet restore command.
-ms.date: 02/27/2020
+ms.date: 05/15/2023
```
Comment (BillWagner) [inline docs/core/tools/dotnet-restore.md:4]: same as earlier comment...

File: docs/core/whats-new/dotnet-8.md [hunk 3]
```diff
@@ -464,11 +699,24 @@ In previous .NET versions, you could build .NET from source, but it required you
...
+
+### .NET blog
+
+- [Announcing .NET 8 Preview 1](https://devblogs.microsoft.com/dotnet/announcing-dotnet-8-preview-1/)
+- [Announcing .NET 8 Preview 2](https://devblogs.microsoft.com/dotnet/announcing-dotnet-8-preview-2/)
+- [Announcing .NET 8 Preview 3](https://devblogs.microsoft.com/dotnet/announcing-dotnet-8-preview-3/)
+- [Announcing .NET 8 Preview 4](https://devblogs.microsoft.com/dotnet/announcing-dotnet-8-preview-4/)
+- [ASP.NET Core updates in .NET 8 Preview 1](https://devblogs.microsoft.com/dotnet/asp-net-core-updates-in-dotnet-8-preview-1/)
+- [ASP.NET Core updates in .NET 8 Preview 2](https://devblogs.microsoft.com/dotnet/asp-net-core-updates-in-dotnet-8-preview-2/)
+- [ASP.NET Core updates in .NET 8 Preview 3](https://devblogs.microsoft.com/dotnet/asp-net-core-updates-in-dotnet-8-preview-3/)
```
Comment (BillWagner) [inline docs/core/whats-new/dotnet-8.md:722]: Consider reversing the order so the most recent update is displayed first:

```suggestion
### .NET blog

- [Announcing .NET 8 Preview 4](https://devblogs.microsoft.com/dotnet/announcing-dotnet-8-preview-4/)
- [Announcing .NET 8 Preview 3](https://devblogs.microsoft.com/dotnet/announcing-dotnet-8-preview-3/)
- [Announcing .NET 8 Preview 2](https://devblogs.microsoft.com/dotnet/announcing-dotnet-8-preview-2/)
- [Announcing .NET 8 Preview 1](https://devblogs.microsoft.com/dotnet/announcing-dotnet-8-preview-1/)
- [ASP.NET Core updates in .NET 8 Preview 3](https://devblogs.microsoft.com/dotnet/asp-net-core-updates-in-dotnet-8-preview-3/)
- [ASP.NET Core updates in .NET 8 Preview 2](https://devblogs.microsoft.com/dotnet/asp-net-core-updates-in-dotnet-8-preview-2/)
- [ASP.NET Core updates in .NET 8 Preview 1](https://devblogs.microsoft.com/dotnet/asp-net-core-updates-in-dotnet-8-preview-1/)
```

**Review by BillWagner — APPROVED** [review]
This looks great @gewarren 

I had a couple optional suggestions. You can :shipit: when you're ready.

File: docs/core/tools/dotnet-build.md (hunk 1 above)
Comment (gewarren) [inline docs/core/tools/dotnet-build.md:4]:
```suggestion
ms.date: 05/16/2023
```

**Pull request merged**